### PR TITLE
[TRAFODION-353][TRAFODION-1200][TRAFODION-1214][TRAFODION-12]

### PR DESCRIPTION
**Pull request by robertamarton**
Privilege manager fixes and feature updates including LDAP 
configuration changes, column level privilege updates, and 
DB__ROOTROLE updates.

### CODE REVIEW

**Comment by sureshsubbiah** [comment]
Was the previous fix for TRAFODION-353 partial? I hope I did not make a mistake in closing JIRA TRAFODION-353 based on its LaunchPad status. Please reopen excuse my mistake if necessary.

File: core/sql/regress/catman1/EXPECTED129 [hunk 1]
```diff
@@ -258,19 +258,16 @@ End of MXCI Session
 >>-- As user1, should succeed
 >>update us4.t1 set col3 = 2 ;
 
-*** ERROR[4481] The user does not have SELECT privilege on table or view TRAFODION.US4.T1.
-
-*** ERROR[8822] The statement was not prepared.
-
+--- 4 row(s) updated.
```
Comment (sureshsubbiah) [inline core/sql/regress/catman1/EXPECTED129]: This issue is documented in TRAFODION 1243. Thank you for fixing it. I am assigning this issue to you.

File: core/sql/sqlcomp/PrivMgr.cpp [hunk 2]
```diff
@@ -894,3 +918,49 @@ void PrivMgr::setFlags()
...
+// Sometimes it is amazing that things actually work with all these levels
+// of interfaces.  Perhaps we can skip a few levels...  
+// ----------------------------------------------------------------------------
+void PrivMgr::log(
+  const std::string filename,
+  const std::string message,
+  const int_32 index)
+{ 
+  // for now, don't log
+  return;
```
Comment (sureshsubbiah) [inline core/sql/sqlcomp/PrivMgr.cpp]: I am curious about this. Why have we disabled logging? Is log4cpp (or is it log4cxx) our current logging framework? Is it working adequately?

**Comment by sureshsubbiah** [comment]
Change looks great. I can merge it by end of today unless someone gets a chance to also look at it.

**Comment by robertamarton** [comment]
I have asked Arvind to take a look at the changes also.



    Roberta



*From:* Suresh Subbiah [mailto:notifications@github.com]
*Sent:* Monday, August 31, 2015 9:23 AM
*To:* apache/incubator-trafodion <incubator-trafodion@noreply.github.com>
*Cc:* Roberta Marton <roberta.marton@esgyn.com>
*Subject:* Re: [incubator-trafodion]
[TRAFODION-353][TRAFODION-1200][TRAFODION-1214][TRAFODION-12] (#69)



Change looks great. I can merge it by end of today unless someone gets a
chance to also look at it.

—
Reply to this email directly or view it on GitHub
<https://github.com/apache/incubator-trafodion/pull/69#issuecomment-136420160>
.

File: core/sql/regress/catman1/EXPECTED129 (hunk 1 above)
Comment (robertamarton) [inline core/sql/regress/catman1/EXPECTED129]: Yes, it looks like we check for select privilege when an update is performed.  The user is not able to update if they don't have select.

File: core/sql/sqlcomp/PrivMgr.cpp (hunk 2 above)
Comment (robertamarton) [inline core/sql/sqlcomp/PrivMgr.cpp]: It is working very well.  But the way it is implemented is to always log messages informational.  I want to change this to use trace instead.  So, we only log if configured with trace.  I had not yet gotten this to work so I decided not to set tracing on yet.

**Comment by robertamarton** [comment]
Once pull request-69  is delivered, it can be closed.  Should I reopen and
then close it once the delivery is complete?



    Roberta



*From:* Suresh Subbiah [mailto:notifications@github.com]
*Sent:* Monday, August 31, 2015 9:03 AM
*To:* apache/incubator-trafodion <incubator-trafodion@noreply.github.com>
*Cc:* Roberta Marton <roberta.marton@esgyn.com>
*Subject:* Re: [incubator-trafodion]
[TRAFODION-353][TRAFODION-1200][TRAFODION-1214][TRAFODION-12] (#69)



Was the previous fix for TRAFODION-353 partial? I hope I did not make a
mistake in closing JIRA TRAFODION-353 based on its LaunchPad status. Please
reopen excuse my mistake if necessary.

—
Reply to this email directly or view it on GitHub
<https://github.com/apache/incubator-trafodion/pull/69#issuecomment-136415800>
.

File: core/sql/sqlcomp/PrivMgrPrivileges.cpp [hunk 3]
```diff
@@ -4407,6 +4609,60 @@ static void getColRowsForGranteeGrantor(
...
+         break;
+      case COM_LIBRARY_OBJECT:
+         if (privBitmap.test(UPDATE_PRIV) && privBitmap.test(USAGE_PRIV))
+            return true;
+         break;      
+      case COM_STORED_PROCEDURE_OBJECT:
+      case COM_USER_DEFINED_ROUTINE_OBJECT:
+         if (privBitmap.test(EXECUTE_PRIV))
+            return true;
+      case COM_SEQUENCE_GENERATOR_OBJECT:
```
Comment (arvind-narain) [inline core/sql/sqlcomp/PrivMgrPrivileges.cpp]: Is a break needed here ?

File: core/sql/sqlcomp/PrivMgr.cpp [hunk 4]
```diff
@@ -894,3 +918,49 @@ void PrivMgr::setFlags()
...
+// Input:
+//    filename - code file that is performing the request 
+//    message  - the message to log
+//    index    - index for logging that loops through a list
+//
+// Background
+//   Privilege manager code sets up a message and calls this log method
+//   This method calls SQLMXLoggingArea::logPrivMgrInfo described in 
+//      sqlmxevents/logmxevent_traf (.h & .cpp)
+//   logPrivMgInfo is a wrapper class around qmscommon/QRLogger (.h & .cpp)
```
Comment (arvind-narain) [inline core/sql/sqlcomp/PrivMgr.cpp]: logPrivMgrInfo - minor - can be touched later.

**Comment by arvind-narain** [comment]
Looks good Roberta. Looked more closely at the ldap changes. Other changes - very minor spelling issues in trace messages ( privlege)

File: core/sql/sqlcomp/PrivMgrPrivileges.cpp (hunk 3 above)
Comment (robertamarton) [inline core/sql/sqlcomp/PrivMgrPrivileges.cpp]: Thanks, it looks like the code is missing a break.  will add it.

**Comment by robertamarton** [comment]
Made changes as suggested by reviewers.

**Comment by robertamarton** [comment]
jenkins, retest

**Pull request merged**